File: zadapter3/src/main/java/com/zone/adapter3/QuickConfig.java
package com.zone.adapter3;

import android.util.Log;

import com.zone.adapter3.loadmore.callback.BaseLoadMoreDelegates;
import com.zone.adapter3.loadmore.callback.ILoadMoreDelegates;
import com.zone.adapter3.loadmore.callback.NullLoadMoreDelegates;


/**
 * Created by Administrator on 2016/3/24.
 */
public class QuickConfig {

    public static final String TAG = "ZAdapter3";

    static boolean writeLog = true;

    ILoadMoreDelegates loadMoreDelegates=new BaseLoadMoreDelegates();


    public ILoadMoreDelegates getLoadMoreDelegates() {
        return loadMoreDelegates;
    }

    public void setiLoadMoreDelegates(ILoadMoreDelegates iLoadMoreDelegates) {
        this.loadMoreDelegates = iLoadMoreDelegates;
    }

    public boolean isWriteLog() {
        return writeLog;
    }

    public void setWriteLog(boolean writeLog) {
        this.writeLog = writeLog;
    }

    public static QuickConfig build() {
        return new QuickConfig();
    }

    public void perform() {
        QuickRcvAdapter.quickConfig = this;
    }


    public static void d(String str) {
        if (writeLog)
            Log.d(TAG, str);
    }

    public static void e(String str) {
        if (writeLog)
            Log.e(TAG, str);
    }

    public static void v(String str) {
        if (writeLog)
            Log.v(TAG, str);
    }

    public static void i(String str) {
        if (writeLog)
            Log.i(TAG, str);
    }
}

File: zadapter3/src/main/java/com/zone/adapter3kt/QuickConfig.kt
package com.zone.adapter3kt

import android.util.Log
import com.zone.adapter3kt.delegate.LoadMoreViewDelegate
import com.zone.adapter3kt.delegate.done.BaseLoadMoreDelegates

import com.zone.adapter3kt.loadmore.LoadingSetting

/**
 * Created by Administrator on 2016/3/24.
 */
class QuickConfig {

    var loadMoreDelegates: LoadMoreViewDelegate = BaseLoadMoreDelegates()
    var loadingSetting: LoadingSetting = LoadingSetting()

    var isWriteLog: Boolean
        get() = writeLog
        set(writeLog) {
            isWriteLog = writeLog
        }

    fun perform() {
        QuickAdapter.quickConfig = this
    }

    companion object {

        val TAG = "ZAdapter3KT"

        internal var writeLog = true
        @JvmStatic
        fun build(): QuickConfig {
            return QuickConfig()
        }
        @JvmStatic
        fun d(str: String) {
            if (writeLog)
                Log.d(TAG, str)
        }

        @JvmStatic
        fun e(str: String) {
            if (writeLog)
                Log.e(TAG, str)
        }
        @JvmStatic
        fun v(str: String) {
            if (writeLog)
                Log.v(TAG, str)
        }
        @JvmStatic
        fun i(str: String) {
            if (writeLog)
                Log.i(TAG, str)
        }
    }
}
